src/balancerd/src/service.rs

```diff
@@ -132,6 +135,18 @@ pub async fn run(args: Args) -> Result<(), anyhow::Error> {
             "exactly one of --static-resolver-addr or --frontegg-resolver-template must be present"
         ),
     };
+    let ticker = IntervalStream::new(tokio::time::interval(Duration::from_secs(60 * 60)));
+    let ticker = ticker.map(|_| {
+        let (tx, rx) = oneshot::channel();
+        mz_ore::task::spawn(|| "balancer ssl reload", async {
+            let Ok(res) = rx.await else { return };
+            if let Err(err) = res {
+                tracing::error!("failed to reload SSL certificate: {err}");
+            }
+        });
+        tx
+    });
```

Comment:
This seems like a lot of trouble to go to just for a `tracing::error!` call. It seems like the `ReloadingContext` could just do the `tracing::error!` directly, and then you wouldn't need the tick stream to emit oneshot senders on which responses from the reloading context were sent back; you could have the tick stream be a very simple stream with an item type of `()`.

Reply:
Moved the logging into the reloading context itself. But the oneshot is still very useful for testing to avoid races. I changed this to be a stream of Option<oneshot..> so that things that don't care can just send None. This simplifies it sufficiently while maintaining testability.

Reply:
Gotcha.